File: 백준/Bronze/2798. 블랙잭/블랙잭.py
from itertools import combinations
import sys

input=sys.stdin.readline

n,m=map(int,input().split())

arr=list(map(int,input().split()))
big_sum=0
for i in combinations(arr,3):
    temp_sum=sum(i)
    if big_sum<temp_sum<=m:
        big_sum=temp_sum
print(big_sum)

File: 백준/Bronze/2798. 블랙잭/블랙잭.kt
fun main() {
    val (n, m) = readLine()!!.split(" ").map { it.toInt() }

    val cards = readLine()!!.split(" ").map { it.toInt() }

    val sortCard = cards.sorted()

    var answer = 0

    for (i in 0 until n) {
        for (j in (i + 1) until n) {
            for (k in (j + 1) until n) {
                val sum = sortCard[i] + sortCard[j] + sortCard[k]
                if (sum > m) {
                    break
                } else if (sum > answer) {
                    answer = sum
                }
            }
        }
    }

    println(answer)
}
